crates/storage/src/schema/revision_0056.rs

```diff
@@ -0,0 +1,127 @@
+        .context("Querying storage_addresses")?;
+    let mut storage_addresses_insert = tx
+        .prepare_cached("INSERT INTO storage_addresses (storage_address) VALUES (?) RETURNING id")
+        .context("Inserting into storage_addresses")?;
+    let mut storage_updates_insert = tx
+    .prepare_cached(
+        "INSERT INTO storage_updates_normalized (block_number, contract_address_id, storage_address_id, storage_value) VALUES (?, ?, ?, ?)",
+    )
+    .context("Inserting into storage_updates_normalized")?;
+    let storage_updates_count = tx
+        .query_row("SELECT COUNT(*) FROM storage_updates", [], |row| {
+            row.get::<_, i64>(0)
+        })
+        .context("Counting storage_updates")?;
+    let mut last_progress = Instant::now();
+    let mut i = 0;
+    loop {
+        let Some((block_number, contract_address, storage_address, storage_value)) =
+            storage_updates.next().transpose()?
+        else {
+            break;
+        };
+        if last_progress.elapsed().as_secs() >= 10 {
+            tracing::info!(
+                "Migrating storage_updates: {:.2}% ({}/{})",
+                i as f64 / storage_updates_count as f64 * 100.0,
+                i,
+                storage_updates_count
+            );
+            last_progress = Instant::now();
+        }
+        let contract_address_id = contract_addresses_query
+            .query_map(params![&contract_address], |row| row.get::<_, i64>(0))
+            .context("Querying contract_addresses")?
+            .next()
+            .unwrap_or_else(|| {
+                contract_addresses_insert
+                    .query_row(params![&contract_address], |row| row.get::<_, i64>(0))
+            })
+            .context("Inserting into contract_addresses")?;
+        let storage_address_id = storage_addresses_query
+            .query_map(params![&storage_address], |row| row.get::<_, i64>(0))
+            .context("Querying storage_addresses")?
+            .next()
+            .unwrap_or_else(|| {
+                storage_addresses_insert
+                    .query_row(params![&storage_address], |row| row.get::<_, i64>(0))
+            })
+            .context("Inserting into storage_addresses")?;
+        storage_updates_insert
+            .execute(params![
+                block_number,
+                contract_address_id,
+                storage_address_id,
+                storage_value
+            ])
+            .context("Inserting into storage_updates_normalized")?;
+        i += 1;
+    }
+
+    tracing::info!(
+        "Migrating storage_updates: {:.2}% ({}/{})",
+        i as f64 / storage_updates_count as f64 * 100.0,
+        i,
+        storage_updates_count
+    );
+
+    tracing::info!("Dropping storage_updates and renaming temporary table");
+
+    tx.execute_batch(
+        r"
+        DROP TABLE storage_updates;
+        ALTER TABLE storage_updates_normalized RENAME TO storage_updates;
```

Comment:
Seems like we're losing the following indices we had on the old `storage_updates` table:

```
CREATE INDEX storage_updates_contract_address_storage_address_block_number ON storage_updates(contract_address, storage_address, block_number);
CREATE INDEX storage_updates_block_number ON storage_updates(block_number);
```

Reply:
It's probably because of these missing indices, but both sync and `starknet_getStateUpdate` is _very_ slow after the migration.

Reply:
Re-creating the indices takes some extra time in addition to the migration (took 2616s for me on mainnet at block 633600):

```
sqlite> CREATE INDEX storage_updates_block_number ON storage_updates(block_number);
Run Time: real 103.829 user 84.027105 sys 18.693363
sqlite> CREATE INDEX storage_updates_contract_address_storage_address_block_number ON storage_updates(contract_address_id, storage_address_id, block_number);
Run Time: real 298.750 user 235.391829 sys 38.971170
```